consensus/src/liveness/proposal_generator.rs

```diff
@@ -168,6 +175,9 @@ pub struct ProposalGenerator {
     // Last round that a proposal was generated
     last_round_generated: Round,
     quorum_store_enabled: bool,
+
+    sys_txn_providers: Vec<Arc<dyn SysTxnProvider>>,
+    should_propose_sys_txns: bool,
```

Comment:
nit: system_txn_enabled?

Reply:
`system_txn_enabled` may sound like if disabled, there's no system txn in the system, which is not true - oracles can still generate system txns as long as its corresponding flag is set.

`should_propose_sys_txns` sounds more accurate.